TimePicker/TimePickerBase.js

```diff
@@ -27,45 +27,33 @@ const hours12 = [
  * @ui
  * @private
  */
-class HourPicker extends Component {
-	static propTypes = {
-		hasMeridiem: PropTypes.bool,
-		value: PropTypes.number
-	};
-
-	constructor (props) {
-		super(props);
-
-		this.state = {
-			noAnimation: false,
-			prevValue: props.value
-		};
-	}
+const HourPicker = (props) => {
+	const [noAnimation, setNoAnimation] = useState(false);
+	const [prevValue, setPrevValue] = useState(props.value);
 
-	static getDerivedStateFromProps (props, state) {
-		if (state.prevValue !== props.value) {
-			const hours = props.hasMeridiem ? hours12 : hours24;
+	useEffect(() => {
+		const hoursArray = props.hasMeridiem ? hours12 : hours24;
 
-			return {
-				noAnimation: hours[state.prevValue] === hours[props.value],
-				prevValue: props.value
-			};
+		if (props.value !== prevValue) {
+			setNoAnimation(hoursArray[prevValue] === hoursArray[props.value]);
+			setPrevValue(props.value);
 		}
-
-		return null;
-	}
-
-	render () {
-		const {hasMeridiem, ...rest} = this.props;
-		const hours = hasMeridiem ? hours12 : hours24;
-
-		return (
-			<DateComponentPicker {...rest} noAnimation={this.state.noAnimation}>
-				{hours}
-			</DateComponentPicker>
-		);
-	}
-}
+	}, [prevValue, props.hasMeridiem, props.value]);
```

Comment:
please change dep array like so
```suggestion
	}, [prevValue, props]);
```